app/controllers/tweeets_controller.rb

```diff
@@ -0,0 +1,75 @@
+class TweeetsController < ApplicationController
+  before_action :set_tweeet, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
**REQUIRED**
* It uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
Let's use a helper method in the ApllicationController to restrict access to `:new` and `:create` since it's a hard requirement.

Reply:
done

Reply:
Why didn't you follow the indication of @wathigo ?